tests/driver/src/test/java/UserTest.java

```diff
@@ -18,14 +21,121 @@ public void selfMethod() {
+                "0.000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000" +
+                        "00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000" +
+                        "00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000" +
+                        "00000000000000000000000000000000000000022250738585072014");
+        assertValueRoundTrip(u::getDouble, u::doubleToString, Double.POSITIVE_INFINITY, "inf");
+        assertValueRoundTrip(u::getDouble, u::doubleToString, Double.NEGATIVE_INFINITY, "-inf");
+    }
+
+    @Test
+    public void longTest() {
+        assertValueRoundTrip(u::getLong, u::longToString, 0L, "0");
+        assertValueRoundTrip(u::getLong, u::longToString, 1L, "1");
+        assertValueRoundTrip(u::getLong, u::longToString, -1L, "-1");
+        assertValueRoundTrip(u::getLong, u::longToString, Long.MAX_VALUE, "9223372036854775807");
+        assertValueRoundTrip(u::getLong, u::longToString, Long.MIN_VALUE, "-9223372036854775808");
+    }
+
+    @Test
+    public void shortTest() {
+        assertValueRoundTrip(u::getShort, u::shortToString, (short) 0, "0");
+        assertValueRoundTrip(u::getShort, u::shortToString, (short) 1, "1");
+        assertValueRoundTrip(u::getShort, u::shortToString, (short) -1, "-1");
+        assertValueRoundTrip(u::getShort, u::shortToString, Short.MAX_VALUE, "32767");
+        assertValueRoundTrip(u::getShort, u::shortToString, Short.MIN_VALUE, "-32768");
+    }
+
+    @Test
+    public void stringTest() {
+        assertValueRoundTrip(u::getString, Function.identity(), "", "");
+        assertValueRoundTrip(u::getString, Function.identity(), "hello!", "hello!");
+        assertValueRoundTrip(u::getString, Function.identity(), "a".repeat(10000), "a".repeat(10000));
+        assertValueRoundTrip(u::getString, Function.identity(), "\0a\rb\nc\t", "\0a\rb\nc\t");
+        assertValueRoundTrip(u::getString, Function.identity(), "아주 좋습니다", "아주 좋습니다");
+        // pirate flag https://unicode.org/emoji/charts/emoji-zwj-sequences.html
+        assertValueRoundTrip(u::getString, Function.identity(), "️🏴‍☠️", "️🏴‍☠️");
+        assertValueRoundTrip(u::getString, Function.identity(), "️️𒅄", "️️𒅄"); // 4 bytes in utf-8
+    }
+
+    @Test
+    public void stringArrayTest() {
+        assertValueRoundTrip(u::getStringArray, u::stringArrayToString, List.of(), "[]");
+        assertValueRoundTrip(u::getStringArray, u::stringArrayToString, List.of("a", "b", "c"), "[\"a\", \"b\", \"c\"]");
     }
 
     @Test
     public void staticMethod() {
         assertEquals(String.valueOf(User.getTotalUsersCount()), User.userCountStatus());
     }
+
+    private <T> void assertValueRoundTrip(Function<T, T> func, Function<T, String> toString, T value, String text) {
+        assertEquals(value, func.apply(value));
+        assertEquals(text, toString.apply(value));
```

Comment:
Decided this was also probably a good idea to make sure that the value rust has is what we expect it to be.
For example, if we converted java booleans to rust bools incorrectly then from rust bools to java booleans incorrectly, we would wind up with the correct java value in the full trip test. This allows us to check what rust has in the middle.